subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/TaskCachingReasonsListener.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks.execution;
+
+import org.gradle.api.Task;
+import org.gradle.api.internal.TaskCaching;
+
+public interface TaskCachingReasonsListener {
+    void taskCacheable(Task task, TaskCaching cacheable);
```

Comment:
This method name doesn't help me understand when this method should be called, or what happens when it gets called. Some Javadoc would help, but the best would be to find a name that describes the role of the method better. Something with a verb would definitely improve matters.